packages/flutter_tools/lib/src/plugins.dart

```diff
@@ -916,7 +916,20 @@ Future<void> _writeMacOSPluginRegistrant(FlutterProject project, List<Plugin> pl
 }
 
 Future<void> _writeWindowsPluginFiles(FlutterProject project, List<Plugin> plugins) async {
-  final List<Map<String, dynamic>> windowsPlugins = _extractPlatformMaps(plugins, WindowsPlugin.kConfigKey);
+  // Plugins without a 'pluginClass' definition shouldn't create native cpp files. The plugin might
+  // be a pure Dart implementation, and the class will be defined by 'dartPluginClass'.
+  final List<Plugin> nativeWindowsPlugins = plugins.where((final Plugin element) {
```

Comment:
Rather that duplicating logic from `_extractPlatformMaps`, how about calling it and then running a `where` over to filter to just those with a `pluginClass`?

Reply:
Added the logic to _filterNativePlugins